### DISCUSSION TITLE
WebAuthn within Web Extensions?

### DISCUSSION THREAD

**Discussion by MaKleSoft** [MasterKale/SimpleWebAuthn#167] [General]
Does anyone have any experience with using WebAuthn within a web extension? It looks like it's not supported from an unpacked extension because those are not considered a secure context, but maybe from an extension loaded from the official store? I haven't gotten around to publishing an extension in order to test this, but maybe somebody else has? Unfortunately I can't find anything about this on the Interwebs...

**Comment by MaKleSoft** [comment]
Ok, now I did find this: https://github.com/w3c/webauthn/issues/1158
Doesn't sound promising. I'm surprised there isn't more discussion around this in general.

**Comment by MasterKale** [comment]
@Mikescops can probably speak most authoritatively to this, see Issue #90. He's endured issues with browser extensions in both Firefox and Chrome.

**Reply by Mikescops** [reply]
Hello, yes sure we're using WebAuthn within web extensions at @Dashlane, I would say that the implementation is far from easy and it's mostly due to the W3C spec being really focused on "websites" implementation so that web browsers just left a half-working code there.

The main issue is here: https://github.com/w3c/webauthn/issues/1372

**Reply by MaKleSoft** [reply]
Thanks @MasterKale for the tip, and thank you @Mikescops for taking the time to respond! Been reading trough the related discussions and it's been really helpful for getting a better understanding of the problem. So you've been able to make it work in Chrome but not in Firefox, correct?

**Reply by Mikescops** [reply]
That's correct, any Chromium browsers are working :)

**Reply by Mikescops** [reply]
I made a demo extension if you want: https://github.com/Mikescops/webauthn-extension-playground

**Reply by MaKleSoft** [reply]
Thanks!
